nixos/modules/services/backup/bupstash.nix

```diff
@@ -0,0 +1,556 @@
+                          type = bool;
+                          default = true;
+                          description = ''
+                            Enable send log for this repository. If set to `false`, the argument `--no-send-log` will be passed to bupstash.
+                            The send log serves two main purposes:
+
+                            -  it remembers the ids of data chunks that were sent to the repository in the last 'put', allowing bupstash to avoid resending those chunks over the network repeatedly.
+                            -  It stores a mapping of file paths to data that has already been sent, allowing bupstash to skip processing files when snapshotting the same directory many times repeatedly.
+                          '';
+                        };
+                      };
+
+                      config = {
+                        command = mkIf (config.kind == "ssh") (
+                          (
+                            s:
+                            "ssh -i ${s.privateKey} ${s.user}@${s.server} ${
+                              optionalString (s.path != null) "bupstash serve ${s.path}"
+                            }"
+                          )
+                            config.ssh
+                        );
+                      };
+                    }
+                  )
+                );
+                default = [ ];
+                description = "List of repositories to backup to.";
+              };
+
+              tags = mkOption {
+                type = attrsOf str;
+                default = {
+                  inherit name;
+                };
+                description = ''
+                  Key/value pairs that are applied as tags for this job.
+                  Valid tag keys must match the regular expression ^([a-zA-Z0-9\\-_]+)=(.+)$,
+                  that means tag keys must be alpha numeric with the addition of - and _
+                '';
+              };
+
+              startAt = mkOption {
+                type = either str (listOf str);
+                default = [ ];
+                description = ''
+                  Automatically start this backup at the given date/time, which
+                  must be in the format described in {manpage}`systemd.time(7)`.
+                '';
+                example = "*-*-* 02:30:00";
+              };
+
+              extraArgs = mkOption {
+                type = listOf str;
+                default = [ ];
+                description = ''
+                  Additional options passed to the `bupstash put` command line.
+                  Check the [documentation](https://bupstash.io/doc/man/bupstash-put.html) for possible options.
+                '';
+              };
+            };
+          }
+        )
+      );
+      default = { };
+      description = "Bupstash backup job configurations.";
+    };
+  };
+
+  config = mkMerge [
+    (mkIf (cfg.jobs != { }) {
+      assertions =
```

Comment:
Add an assertion, that ensures either `paths` or `commands` is configured, never both.